lib/pbench/cli/server/user_management.py

```diff
@@ -0,0 +1,208 @@
+@pass_cli_context
+def user_delete(context: object, username: str) -> None:
+    try:
+        # Delete the the user with specified username
+        config_setup(context)
+        User.delete(username=username)
+        rv = 0
+    except Exception as exc:
+        click.echo(exc, err=True)
+        rv = 2 if isinstance(exc, BadConfig) else 1
+
+    click.get_current_context().exit(rv)
+
+
+# Users list CLI
+@user_command_cli.command()
+@common_options
+@pass_cli_context
+def user_list(context: object) -> None:
+    try:
+        config_setup(context)
+        click.echo(USER_LIST_HEADER_ROW)
+
+        # Query all the users
+        users = User.query(all=True)
+
+        for user in users:
+            click.echo(
+                USER_LIST_ROW_FORMAT.format(
+                    user.username,
+                    user.first_name,
+                    user.last_name,
+                    user.registered_on.strftime("%Y-%m-%d"),
+                    user.email,
+                )
+            )
+
+        rv = 0
+    except Exception as exc:
+        click.echo(exc, err=True)
+        rv = 2 if isinstance(exc, BadConfig) else 1
+
+    click.get_current_context().exit(rv)
+
+
+# User update CLI
+@user_command_cli.command()
+@common_options
+@click.argument("updateuser")
+@click.option(
+    "--username", required=False, help="Specify the new username",
+)
+@click.option(
+    "--email", required=False, help="Specify the new email",
+)
+@click.option(
+    "--first-name", required=False, help="Specify the new first name",
+)
+@click.option(
+    "--last-name", required=False, help="Specify the new last name",
+)
+@click.option(
+    "--role",
+    required=False,
+    type=click.Choice([role.name for role in Roles], case_sensitive=False),
+    help="Specify the new role",
+)
+@pass_cli_context
+def user_update(
+    context: object,
+    updateuser: str,
+    username: str,
+    first_name: str,
+    last_name: str,
+    email: str,
+    role: str,
+) -> None:
+    try:
+        config_setup(context)
+        # Query the user
+        user = User.query(username=updateuser)
+
+        if user is None:
+            click.echo(f"User {updateuser} doesn't exist")
+            rv = 1
+        else:
+            dict_to_update = {}
+            if username:
+                dict_to_update["username"] = username
+
+            if first_name:
+                dict_to_update["first_name"] = first_name
+
+            if last_name:
+                dict_to_update["last_name"] = last_name
+
+            if email:
+                dict_to_update["email"] = email
+
+            if role:
+                dict_to_update["role"] = role
+
+            # Update the user
+            user.update(**dict_to_update)
+
+            click.echo(f"User {updateuser} updated")
+            rv = 0
+    except Exception as exc:
+        click.echo(exc, err=True)
+        rv = 2 if isinstance(exc, BadConfig) else 1
+
+    click.get_current_context().exit(rv)
```

Comment:
OK...I give up:  what changed in this file?

Reply:
one thing changed in the last commit is that, instead of doing `users = Database.db_session.query(User).all()` in `user_list` command I am doing `users = User.query(all=True)`